plugins/network-elements/ovs/src/com/cloud/network/element/OvsElement.java

```diff
@@ -468,19 +470,23 @@ public boolean applyStaticNats(final Network network, final List<? extends Stati
             return true;
         }
 
-        DataCenterVO dcVO = _dcDao.findById(network.getDataCenterId());
-        NetworkTopology networkTopology = _networkTopologyContext.retrieveNetworkTopology(dcVO);
-
-        return networkTopology.applyStaticNats(network, rules, routers);
+        final DataCenterVO dcVO = _dcDao.findById(network.getDataCenterId());
+        final NetworkTopology networkTopology = _networkTopologyContext.retrieveNetworkTopology(dcVO);
+        boolean result = false;
+        for (final DomainRouterVO domainRouterVO : routers) {
+            result = networkTopology.applyStaticNats(network, rules, domainRouterVO);
```

Comment:
Good point. Thought about that when I was coding, but not on the Ovs one. If you look, I did the same on other places. To be honest, I just forgot to change. When I changed first time and thought  about that, I just gave it a run because I was trying to narrow down the problem. 6 hours later, when I got it working, I forgot to change.

I can change that tomorrow... os later tonight. Or we can get this one through and fix it on Monday, since the PR/Feature will be already in. I just want to get through the feature release.

Spent the whole day behind the mac. If I do 30min tomorrow, wife will kill me. :D